tests/helpers/watch-property.js

```diff
@@ -0,0 +1,144 @@
+import Ember from 'ember';
+import QUnit from 'qunit';
+
+const {
+  addObserver,
+  removeObserver
+} = Ember;
+
+function makeCounter() {
+  let count = 0;
+  const counter = Object.create(null);
+  counter.reset = function resetCounter() { count = 0; };
+
+  Object.defineProperty(counter, 'count', {
+    get() { return count; },
+    set() {},
+    configurable: false,
+    enumerable: true
+  });
+
+  Object.freeze(counter);
+
+  function increment() {
+    count++;
+  }
+
+  return { counter, increment };
+}
```

Comment:
@hjdivad because I wanted to be able to have a strong guarantee that nothing else mutated the counter. Notice how the counter is frozen and the property is a read-only property that closes over a private var.

Reply:
Okay.  I think it's a bit overkill for our own tests (as opposed to say, a public API) but the code is clear enough that either way is fine with me.